ui/app/AppLayouts/Profile/popups/SearchEngineModal.qml

```diff
@@ -21,6 +21,13 @@ StatusDialog {
     horizontalPadding: Theme.halfPadding
     verticalPadding: Theme.halfPadding
 
+    Component.onCompleted: {
+        if (!SearchEnginesConfig.getEngineById(accountSettings.selectedBrowserSearchEngineId)) {
+            console.warn("SearchEngineModal: Invalid search engine ID detected, resetting to DuckDuckGo")
+            accountSettings.selectedBrowserSearchEngineId = SearchEnginesConfig.browserSearchEngineDuckDuckGo
```

Comment:
Does that mean the invalid search engine will be reset only when going to Settings? What if the user already had configured it that way before, and they do the search w/o going to Settings?